### PR TITLE
add doctests for DETR

### PR DESCRIPTION
**Pull request by qherreros**
# What does this PR do?

Enable doctests for DETR

@patrickvonplaten

### CODE REVIEW

File: src/transformers/models/detr/modeling_detr.py [hunk 1]
```diff
@@ -1397,8 +1399,15 @@ def forward(
         >>> inputs = feature_extractor(images=image, return_tensors="pt")
         >>> outputs = model(**inputs)
         >>> # model predicts bounding boxes and corresponding COCO classes
-        >>> logits = outputs.logits
-        >>> bboxes = outputs.pred_boxes
+        >>> bboxes, logits = outputs.pred_boxes, outputs.logits
+
+        >>> probas_per_class = outputs.logits.softmax(-1)[:, :, :-1]
```
Comment (ydshieh) [inline src/transformers/models/detr/modeling_detr.py:1404]: I guess the last one is for `no-object`? Maybe a comment here would help the reader to understand the doc. easier 😄

**Review by ydshieh — APPROVED** [review]
Thank you @qherreros for working on this. LGTM, and I just left a nit to make things easier to understand for the doc. readers.

**Review by NielsRogge — APPROVED** [review]
Thanks for working on this!

**Comment by patrickvonplaten** [comment]
Good to merge for me!

**Pull request merged**